### ISSUE TITLE
sigmoid vs. softmax

### ISSUE THREAD

**Issue by ahmedabbas81** [shap/shap#1048]
Dear all,

I am making a multiclass classification deep neural network model. It is fully connected neural network. I know I should use softmax at the output layer.

However, when I use softmax, I get non stable (change if I train the model several times) cotributions of inputs to outputs. I can say that I get strange results. The same feature may be contributing strongly to several outputs, although it should contribute strongly to a single output only

If I use sigmoid and select the output with the highest score, I get very good results

Is there a reason behind the change in results between softmax and sigmoid?

And is it wrong to use sigmoid in my case (multi class classification) and select the output with highest score?

Thank you

**Comment by dtemko** [comment]
Hi,

I have the same question
